components/ProductInfo/middle/ShirtOptions.js

```diff
@@ -12,57 +13,83 @@ const ShirtOptions = () => {
   const [see, setSee] = useState(false);
 
   const dispatch = useDispatch();
-  const displayImg = useSelector(({ prodInfo }) => prodInfo.image);
-  const currentDisplay = displayImg;
+  const { currentImage, imageDisplayBtnArr } = useSelector(({ prodInfo }) => ({
+    currentImage: prodInfo.currentImage,
+    imageDisplayBtnArr: prodInfo.imageBtnArr,
+  }));
 
   const mouseEnter = (newImage) => {
-    dispatch(changeVisibleImage(newImage));
+    currentImage.isProd && dispatch(changeVisibleImage(newImage));
   };
```

Comment:
if the currentImage is the actual shirt, then change the display image to the new color shirt